cdap-system-app-api/src/main/java/io/cdap/cdap/api/service/worker/RemoteExecutionException.java

```diff
@@ -23,7 +23,14 @@
  */
 public class RemoteExecutionException extends Exception {
 
+  private final RemoteTaskException cause;
+
   public RemoteExecutionException(RemoteTaskException cause) {
     super(cause.getMessage(), cause);
+    this.cause = cause;
+  }
+
+  public RemoteTaskException getRemoteTaskException() {
```

Comment:
Why need this? Shouldn't the `getCause()` method covered this?

Reply:
Was trying to avoid type cast to get RemoteTaskException. RemoteTaskException is useful for getting the getRemoteExceptionClassName()

Reply:
You can override the getCause to return a more specific type

Reply:
Done